main.py

```diff
@@ -298,34 +259,61 @@ def status(message):
     LOGGER.info("Hostname: %s", hostname)
 
     # Start reading from sensors
-    sensor_thread = Thread(target=read_data,
-                           args=(output_sensors, input_sensors, queue))
+    sensor_thread = Thread(target=read_data, args=(output_sensors, input_sensors, queue))
     sensor_thread.start()
-
-    # Start server
-    LOGGER.info("Starting server")
-    server = CoAPServer(("224.0.1.187", 5683), multicast=True)
-    server.add_resource('data/', DataResource(queue, input_sensors))
-    server.add_resource('name={}'.format(hostname), DummyResource())
-
+    
+    #load config file
+    with open("config.yml", 'r') as ymlfile:
+        cfg = yaml.load(ymlfile)
+
+    # Create mqtt client
+    client = paho.Client(cfg['mqtt']['ClientId'],cfg['mqtt']['ClSes'],cfg['mqtt']['udata'])
+    client.username_pw_set(username=cfg['mqtt']['uname'],password=cfg['mqtt']['password'])
+    #define callabcks
+    client.on_connect=on_connect
+    client.on_publish = on_publish
+    client.on_disconnect=on_disconnect
+    client.on_log=on_log
+    #reconnect interval on disconnect
+    client.reconnect_delay_set(3)
+    #establish client connection
+    while True:
+        try:
+            client.connect(cfg['mqtt']['server'],cfg['mqtt']['port'])
+            LOGGER.info("Client connected successfully to broker")
+            break
+        except:
+            LOGGER.exception("Connection failure...trying to reconnect...")
+            time.sleep(15)
+    client.loop_start()
+    #continuosly get data from queue and publish to broker
     while True:
         try:
-            server.listen()
+            data=queue.peek(blocking=True)
+            LOGGER.debug("Publishing:"+str(data))
+            info=client.publish("prism/monitorA002",str(data) , qos=1)
```

Comment:
I don't think we want to do `str(data)` -- it might have some weird effects. `json.dumps(data)` is the way to go. You mentioned you were running into an error using this method. Do you know what the error message was?

Reply:
I was looking through the deleted code above and I've run into this issue before. Here is the code I used to fix it (you might have to modify it slightly):

```
data = {k.decode: (value.decode() if isinstance(value, bytes) else value, unit.decode()) 
        for k,v in data.items()}
```

This takes care of converting the keys, values, and units from bytes into strings so that `json.dumps(data)` shouldn't raise an exception.